x/evm/artela/contract/contract_func.go

```diff
@@ -263,14 +262,18 @@ func (k *AspectNativeContract) CheckIsAspectOwnerByCode(ctx sdk.Context, aspectI
 	return binding, runErr
 }
 
-func (k *AspectNativeContract) deploy(ctx sdk.Context, aspectId common.Address, code []byte, properties []types.Property) (*evmtypes.MsgEthereumTxResponse, error) {
-	k.aspectService.aspectStore.StoreAspectCode(ctx, aspectId, code)
-	k.aspectService.aspectStore.StoreAspectProperty(ctx, aspectId, properties)
-
-	level, err := k.checkBlockLevel(aspectId, code)
+func (k *AspectNativeContract) deploy(ctx sdk.Context, aspectId common.Address, code []byte, properties []types.Property, joinPoint *big.Int) (*evmtypes.MsgEthereumTxResponse, error) {
+	aspectVersion := k.aspectService.aspectStore.StoreAspectCode(ctx, aspectId, code)
+	err := k.aspectService.aspectStore.StoreAspectJP(ctx, aspectId, aspectVersion, joinPoint)
 	if err != nil {
 		return nil, err
 	}
+	propErr := k.aspectService.aspectStore.StoreAspectProperty(ctx, aspectId, properties)
```

Comment:
No need to name so many `xxxErr` in your code, you can just use one `err` for simplicity. Unless you need to distinguish more than 1 errors in your code.